src/stcal/jump/jump.py

```diff
@@ -675,21 +707,33 @@ def find_faint_extended(indata, gdq, readnoise_2d, nframes, snr_threshold=1.3,
     data[gdq == 1] = np.nan
     data[gdq == jump_flag] = np.nan
     all_ellipses = []
-    for intg in range(data.shape[0]):
-        diff = np.diff(data[intg], axis=0)
-        median_diffs = np.nanmedian(diff, axis=0)
+    first_diffs = np.diff(data, axis=1)
+    first_diffs_masked = np.ma.masked_array(first_diffs, mask=np.isnan(first_diffs))
+    nints = data.shape[0]
+    if nints > minimum_sigclip_groups:
+        mean, median, stddev = stats.sigma_clipped_stats(first_diffs_masked, sigma=5,
+                                                         axis=0)
+    for intg in range(nints):
         # calculate sigma for each pixel
-        sigma = np.sqrt(np.abs(median_diffs) + read_noise_2 / nframes)
-
-        # The difference from the median difference for each group
-        e_jump = diff - median_diffs[np.newaxis, :, :]
-
-        ratio = np.abs(e_jump) / sigma[np.newaxis, :, :]  # SNR ratio of
-        # each diff.
+        if nints <= minimum_sigclip_groups:
+            median_diffs = np.nanmedian(first_diffs_masked[intg], axis=0)
+            sigma = np.sqrt(np.abs(median_diffs) + read_noise_2 / nframes)
+           # The difference from the median difference for each group
```

Comment:
Looks like the indentation of the 2 comment lines in this block doesn't match the rest of the block. Looks like they need 1 more leading whitespace. codestyle might complain about that.